core/src/test/scala/unit/kafka/server/KafkaConfigTest.scala

```diff
@@ -766,6 +768,86 @@ class KafkaConfigTest {
     })
   }
 
+  @Test
+  def testDynamicLogConfigs(): Unit = {
+    def getBaseProperties(): Properties = {
+      val validRequiredProperties = new Properties()
+      validRequiredProperties.put(KafkaConfig.ZkConnectProp, "127.0.0.1:2181")
+      validRequiredProperties
+    }
+
+    val props = getBaseProperties()
+    val config = KafkaConfig.fromProps(props)
+
+    def assertDynamic(property: String, value: Any, accessor: () => Any): Unit = {
+      val initial = accessor()
+      props.put(property, value.toString)
+      config.updateCurrentConfig(new KafkaConfig(props))
+      assertNotEquals(initial, accessor())
+    }
+
+    // Test dynamic log config values can be correctly passed through via KafkaConfig to LogConfig
+    // Every log config prop must be explicitly accounted for here.
+    // A value other than the default value for this config should be set to ensure that we can check whether
+    // the value is dynamically updatable.
+    LogConfig.configNames.foreach {
```

Comment:
This is nice! Would it be better to consider the configs defined in `LogConfig.TopicConfigSynonyms` only? Maybe something like this:

```
    LogConfig.TopicConfigSynonyms.foreach { case (logConfig, kafkaConfig) =>
        logConfig match {
          case LogConfig.CleanupPolicyProp =>
            assertDynamic(kafkaConfig, Defaults.Compact, () => config.logCleanupPolicy)
          ...
        }
    }
```